csharp/src/Ice/VectoredBuffer.cs

```diff
@@ -0,0 +1,549 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Ice
+{
+    /// <summary>Represents a position if a VectoredBuffer, the position is compose of
+    /// the segment index and the relative offset in the segment.</summary>
+    public struct BufferPosition
```

Comment:
Since this position is specific to VectoredBuffer, I think it would make sense to nest it inside VectoredBuffer, with:
```
public class VectoredBuffer
{
    public struct Position { ... }
    Position Pos { get {...} set {...} }
}
```